test/unit/ajax-test.js

```diff
@@ -221,6 +224,8 @@ define([
                 done();
             }
         });
+        server.respond();
```

Comment:
We're not just testing various response handlers, we're testing integration with XHR. This replaces XHR with fake XHR. That limits the code coverage we get because it polyfills the browsers the tests run in.